libr/util/buf.c

```diff
@@ -402,7 +402,10 @@ R_API RBuffer *r_buf_new() {
 }
 
 R_API const ut8 *r_buf_buffer (RBuffer *b) {
-	return (b && !b->sparse)? b->buf: NULL;
+	if (b && !b->sparse && b->fd == -1 && !b->mmap) {
+		return b->buf;
+	}
+	r_return_val_if_fail (false, NULL);
```

Comment:
Mmm why? Is it a programming error if one calls r_buf_buffer on a sparse RBuffer? This change would imply one has to know the internal details of the RBuffer before calling this function and it seems to go against the purpose of having a general RBuffer